### PR TITLE
Overwriting files in rules and decoders

### PR DESCRIPTION
**Pull request by yenienserrano**
**Description:**

When trying to upload a file with the same name it overwrote it and did not warn you that a file existed. What I did is to recover a functionality that was lost that tells you which files were uploaded or not if any were not uploaded, and add a checkbox for the option to overwrite or not.

**Test:**

1. Go to "Management/Rules or Decoders"
2. Click on "Manage rules files" or "Manage decoders files"
3. Click on "Import files"
4. import files with or without the overwrite option

**Issue:** 

- #4056 

**Screenshot:**

https://user-images.githubusercontent.com/63758389/171361459-6071fe7b-e894-4167-8429-90d6ac85d27e.mp4

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 # Change Log
 
 All notable changes to the Wazuh app project will be documented in this file.
+## Wazuh v4.3.4 - Kibana 7.10.2, 7.16.x, 7.17.x - Revision 4303
```
Comment (AlexRuiz7) [inline CHANGELOG.md:4]: Revision 4305

File: public/controllers/management/components/management/ruleset/actions-buttons.js [hunk 2]
```diff
@@ -93,15 +93,15 @@ class WzRulesetActionButtons extends Component {
...
+  async uploadFiles(files, resource, overwrite) {
     try {
       let errors = false;
       let results = [];
       const rulesetHandler = new RulesetHandler(resource);
       for (let idx in files) {
         const { file, content } = files[idx];
         try {
-          await rulesetHandler.updateFile(file, content, resource !== RulesetResources.LISTS); // True does not overwrite the file
+          await rulesetHandler.updateFile(file, content, overwrite); // True does not overwrite the file
```
Comment (AlexRuiz7) [inline public/controllers/management/components/management/ruleset/actions-buttons.js:104]: Does that comment make sense?

File: public/controllers/management/components/upload-files.js [hunk 3]
```diff
@@ -37,10 +39,15 @@ export class UploadFiles extends Component {
...
       isPopoverOpen: false,
       uploadErrors: false,
-      errorPopover: false
+      errorPopover: false,
+      overwrite: false
     };
     this.maxSize = 10485760; // 10Mb
   }
+  
+  activeOverwrite(e) {
```
Comment (AlexRuiz7) [inline public/controllers/management/components/upload-files.js:48]: Rename to `setOverwrite()`

File: public/controllers/management/components/upload-files.js [hunk 4]
```diff
@@ -83,26 +90,38 @@ export class UploadFiles extends Component {
...
-                        <li
-                          key={`ruleset-imported-file-${f.file}`}
-                          style={{ listStyle: 'circle' }}
-                        >
-                          {f.file}
-                        </li>
-                      ))}
-                    </ul>
-                  </Fragment>
+                const noUploaded = [];
```
Comment (AlexRuiz7) [inline public/controllers/management/components/upload-files.js:93]: What does `noUpload` stands for? Files not uploaded due to error?

If so, rename the variable to something more descriptive, as `failedUploads` or `notUploadedFiles`.

File: public/controllers/management/components/upload-files.js [hunk 5]
```diff
@@ -284,6 +303,13 @@ export class UploadFiles extends Component {
           <EuiTitle size="m">
             <h1>{`Upload ${this.props.resource}`}</h1>
           </EuiTitle>
+          <EuiCheckbox
+            className='wz-margin-top-10 wz-margin-bottom-10'
+            id="overwrite"
+            label="Activate overwriting"
```
Comment (AlexRuiz7) [inline public/controllers/management/components/upload-files.js:309]: Change the label to `Overwrite`.

File: public/controllers/management/components/upload-files.js [hunk 6]
```diff
@@ -284,6 +303,13 @@ export class UploadFiles extends Component {
           <EuiTitle size="m">
             <h1>{`Upload ${this.props.resource}`}</h1>
           </EuiTitle>
+          <EuiCheckbox
```
Comment (AlexRuiz7) [inline public/controllers/management/components/upload-files.js:306]: Suggestion: use a [Switch](https://elastic.github.io/eui/#/forms/selection-controls#switch) instead.

File: public/controllers/management/components/upload-files.js [hunk 7]
```diff
@@ -284,6 +303,13 @@ export class UploadFiles extends Component {
...
+            className='wz-margin-top-10 wz-margin-bottom-10'
+            id="overwrite"
+            label="Activate overwriting"
+            checked={this.state.overwrite}
+            onChange={(e) => this.activeOverwrite(e)}
```
Comment (AlexRuiz7) [inline public/controllers/management/components/upload-files.js:311]: Would `onChange={this.activeOverwrite}` work?

File: public/controllers/management/components/upload-files.js [hunk 8]
```diff
@@ -318,12 +344,26 @@ export class UploadFiles extends Component {
...
+                        >
+                          Close
+                        </EuiButtonEmpty>
+                      </EuiFlexItem>
+                      <EuiFlexItem grow={false}>
+                        <EuiButton
+                          className="upload-files-button"
+                          fill
+                          iconType="sortUp"
+                          onClick={() => this.startUpload()}
```
Comment (AlexRuiz7) [inline public/controllers/management/components/upload-files.js:361]: Would `onClick={this.startUpload}` work?

**Review by AlexRuiz7 — CHANGES_REQUESTED** [review]

File: public/controllers/management/components/management/ruleset/actions-buttons.js (hunk 2 above)
Comment (Desvelao) [reply public/controllers/management/components/management/ruleset/actions-buttons.js:104]: It seems that the comment has no sense. We could remove it.

File: public/controllers/management/components/upload-files.js (hunk 7 above)
Comment (Desvelao) [reply public/controllers/management/components/upload-files.js:311]: If the logic of that method, uses the `this` keyword, we could put:

```
onChange={this.activeOverwrite}
```

but it is required to bind the `this` context to the method. In the class constructor, we should put:
```
this.activeOverwrite = this.activeOverwrite.bind(this);
```

File: public/controllers/management/components/management/ruleset/actions-buttons.js (hunk 2 above)
Comment (AlexRuiz7) [reply public/controllers/management/components/management/ruleset/actions-buttons.js]: Solved

File: public/controllers/management/components/upload-files.js (hunk 4 above)
Comment (AlexRuiz7) [reply public/controllers/management/components/upload-files.js]: Solved

File: public/controllers/management/components/upload-files.js (hunk 8 above)
Comment (AlexRuiz7) [reply public/controllers/management/components/upload-files.js]: Solved

File: public/controllers/management/components/upload-files.js (hunk 7 above)
Comment (AlexRuiz7) [reply public/controllers/management/components/upload-files.js]: Solved

**Review by AlexRuiz7 — APPROVED** [review]
- Code Review: ✔️ 
- Test: ✔️

**Review by Desvelao — APPROVED** [review]
**review**
code review :heavy_check_mark: 
test :heavy_check_mark:

**Pull request merged**